### PR TITLE
VIVI-2930 add engine factory

### PR DESCRIPTION
**Pull request by drmarkfernie**
*Note*: the `socket.io.js` file is the generated output of `make socket.io.js`, and should not be manually modified.

### The kind of change this PR does introduce

* [ ] a bug fix
* [x] a new feature
* [ ] an update to the documentation
* [ ] a code change that improves performance
* [ ] other

### New behaviour
add factory function to Manager opts to allow alternative Engine.io sockets to be injected.

### CODE REVIEW

File: lib/manager.js [hunk 1]
```diff
@@ -65,6 +65,7 @@ function Manager (uri, opts) {
   var _parser = opts.parser || parser;
   this.encoder = new _parser.Encoder();
   this.decoder = new _parser.Decoder();
+  this.engineFactory = opts.engineFactory || defaultEngineFactory;
```
Comment (tristanpenman) [inline lib/manager.js:68]: Something to consider... You could avoid the need for the `defaultEngineFactory` function by changing this to:
```
opts.engineFactory || eio
```

Comment (DuBistKomisch) [reply lib/manager.js:68]: or at least call `new`

**Comment by DuBistKomisch** [comment]
I don't think we care about any of the dist files

**Review by DuBistKomisch — APPROVED** [review]
fine otherwise though

File: lib/manager.js (hunk 1 above)
Comment (drmarkfernie) [reply lib/manager.js:68]: going to leave as is, we'll worry about changes when I try to merge upstream.

**Pull request merged**